superset/utils/webdriver.py

```diff
@@ -0,0 +1,190 @@
+from typing import Any, Callable, Dict, List, Optional, Tuple, TYPE_CHECKING
+
+from flask import current_app, request, Response, session
+from flask_login import login_user
+from retry.api import retry_call
+from selenium.common.exceptions import TimeoutException, WebDriverException
+from selenium.webdriver import chrome, firefox
+from selenium.webdriver.common.by import By
+from selenium.webdriver.remote.webdriver import WebDriver
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.webdriver.support.ui import WebDriverWait
+from werkzeug.http import parse_cookie
+
+from superset.utils.urls import headless_url
+
+WindowSize = Tuple[int, int]
+logger = logging.getLogger(__name__)
+
+# Time in seconds, we will wait for the page to load and render
+SELENIUM_CHECK_INTERVAL = 2
+SELENIUM_RETRIES = 5
+SELENIUM_HEADSTART = 3
+
+
+if TYPE_CHECKING:
+    # pylint: disable=unused-import
+    from flask_appbuilder.security.sqla.models import User
+
+
+def get_auth_cookies(user: "User") -> List[Dict[Any, Any]]:
+    # Login with the user specified to get the reports
+    with current_app.test_request_context("/login"):
+        login_user(user)
+        # A mock response object to get the cookie information from
+        response = Response()
+        current_app.session_interface.save_session(current_app, session, response)
+
+    cookies = []
+
+    # Set the cookies in the driver
+    for name, value in response.headers:
+        if name.lower() == "set-cookie":
+            cookie = parse_cookie(value)
+            cookies.append(cookie["session"])
+    return cookies
+
+
+def auth_driver(
+    driver: WebDriver,
+    user: "User",
+    auth_cookies_func: Optional[Callable[["User"], List[Dict[Any, Any]]]] = None,
+) -> WebDriver:
+    """
+        Default AuthDriverFuncType type that sets a session cookie flask-login style
+    :return: WebDriver
+    """
+
+    if user:
+        # Set the cookies in the driver
+        for cookie in (auth_cookies_func or get_auth_cookies)(user):
+            info = dict(name="session", value=cookie)
+            driver.add_cookie(info)
+    elif request.cookies:
+        cookies = request.cookies
+        for k, v in cookies.items():
+            cookie = dict(name=k, value=v)
+            driver.add_cookie(cookie)
+    return driver
+
+
+class WebDriverProxy:
+    def __init__(
+        self,
+        driver_type: str,
+        window: Optional[WindowSize] = None,
+        auth_func: Optional[
+            Callable[..., Any]
+        ] = None,  # pylint: disable=bad-whitespace
+        get_cookies_func: Optional[
+            Callable[..., Any]
+        ] = None,  # pylint: disable=bad-whitespace
+    ):
+        self._driver_type = driver_type
+        self._window: WindowSize = window or (800, 600)
+
+        config_auth_func = current_app.config["WEBDRIVER_AUTH_FUNC"] or auth_driver
+        self._auth_func = auth_func or config_auth_func
+
+        config_auth_cookies_func = (
```

Comment:
allowing for potential overriding of `auth` get_cookies func

Reply:
ok, but don't see much value in it, note: `info = dict(name="session", value=cookie)` is done by `auth_driver`